testing/framework/TestCommon.py

```diff
@@ -482,34 +486,44 @@ def must_exist_one_of(self, files):
         print("Missing one of: `%s'" % "', `".join(missing))
         self.fail_test(missing)
 
-    def must_match(self, file, expect, mode = 'rb', match=None, message=None, newline=None):
+    def must_match(self, fname, expect, mode='rb', match=None, message=None, newline=None):
         """Matches the contents of the specified file (first argument)
         against the expected contents (second argument).  The expected
         contents are a list of lines or a string which will be split
         on newlines.
         """
-        file_contents = self.read(file, mode, newline)
+        file_contents = self.read(fname, mode, newline)
+        if 'b' in mode:
+            # Python 3: reading a file in binary mode returns a
+            # bytes object. We cannot find the index of a different
+            # (str) type in that, so convert.
+            file_contents = to_str(file_contents)
```

Comment:
Was this breaking one(or more) of the test framework tests?

Reply:
I think that was the motivation, but it's been a while. Maybe tomorrow I can run the tests both ways and see.

Reply:
there is indeed (one) failure in the old way, the `test_failure` test of `must_contain_TestCase`